src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.ToDoCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * Parser deals with making sense of the user command.
+ *
+ * @author Samsation
+ * @version CS2103T AY 22/23 Sem 1
+ *
+ */
+
+public class Parser {
+
+    private enum CommandTag {
+        BYE, LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND
+    }
+
+    /**
+     * A method that creates a new Command according to the user input.
+     *
+     * @param inputCommand Input command from the user.
+     * @return A specific-typed Command corresponding to the user input.
+     * @throws DukeException If the user input is unrecognised.
+     */
+    public static Command parse(String inputCommand) throws DukeException {
+        try {
+            String[] splitInputCommand = inputCommand.trim().split(" ", 2);
+            CommandTag ct = CommandTag.valueOf(splitInputCommand[0].toUpperCase());
```

Comment:
According to the coding standards:
> scratch variables used for temporary storage or indices can be kept short

I'm not sure if `ct` here fits the criteria, or if it should be changed to something more meaningful like `commandTag`. What do you think?